charts/zabbix/templates/service.yaml

```diff
@@ -10,27 +10,27 @@ metadata:
     helm.sh/chart: {{ include "zabbix.chart" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}-zabbix-server
     app.kubernetes.io/managed-by: {{ .Release.Service }}-zabbix-server
-  {{- if .Values.zabbixserver.service.annotations }}
+  {{- if .Values.zabbixServer.service.annotations }}
   annotations:
-    {{- range $key,$value := .Values.zabbixserver.service.annotations }}
+    {{- range $key,$value := .Values.zabbixServer.service.annotations }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
   {{- end }}
 spec:
-  type: {{ .Values.zabbixserver.service.type }}
-  {{- if .Values.zabbixserver.service.externalTrafficPolicy }}
-  externalTrafficPolicy: {{ .Values.zabbixserver.service.externalTrafficPolicy }}
+  type: {{ .Values.zabbixServer.service.type }}
+  {{- if .Values.zabbixServer.service.externalTrafficPolicy }}
+  externalTrafficPolicy: {{ .Values.zabbixServer.service.externalTrafficPolicy }}
   {{- end }}
-  {{- if .Values.zabbixserver.service.clusterIP }}
-  clusterIP: {{ .Values.zabbixserver.service.clusterIP }}
+  {{- if .Values.zabbixServer.service.clusterIP }}
+  clusterIP: {{ .Values.zabbixServer.service.clusterIP }}
   {{- end }}
   ports:
-    - port: {{ .Values.zabbixserver.service.port }}
+    - port: {{ .Values.zabbixServer.service.port }}
       targetPort: 10051
       name: zabbix-server
       protocol: TCP
-      {{- if and .Values.zabbixserver.service.nodePort (eq .Values.zabbixserver.service.type "NodePort") }}
-      nodePort: {{ .Values.zabbixserver.service.nodePort }}
+      {{- if ( and (eq .Values.zabbixServer.service.type "NodePort" ) (not (empty .Values.zabbixServer.service.nodePort)) ) }}
+      nodePort: {{ .Values.zabbixServer.service.nodePort }}
       {{- end }}
     - port: 10052
```

Comment:
I'd remove port 10052 here as anyway nothing is listening on this port. Though it does not hurt to have it here too.